crates/verify/src/utils.rs

```diff
@@ -136,6 +137,56 @@ pub fn build_using_cache(
     eyre::bail!("couldn't find cached artifact for contract {}", args.contract.name)
 }
 
+pub fn get_immutable_refs(
+    artifact: &CompactContractBytecode,
+) -> Option<BTreeMap<String, Vec<Offsets>>> {
+    if artifact.deployed_bytecode.as_ref().is_some_and(|b| !b.immutable_references.is_empty()) {
+        let immutable_refs =
+            artifact.deployed_bytecode.as_ref().unwrap().immutable_references.clone();
+
+        return Some(immutable_refs);
+    }
+
+    None
+}
+
+pub fn extract_immutables_refs(
+    immutable_refs: BTreeMap<String, Vec<Offsets>>,
+    mut bytecode: Bytes,
+) -> Bytes {
+    let mut total_len_extracted_expected = 0;
+    let init_length = bytecode.len();
+    for (_key, offsets) in immutable_refs {
+        for offset in offsets {
+            let start = offset.start as usize;
+            let end = (offset.start + offset.length) as usize;
+
+            total_len_extracted_expected += offset.length;
+
+            // Remove this sections of bytes from the bytecode
+            let start_section = bytecode.slice(0..start);
+            let end_section = bytecode.slice(end..bytecode.len());
+
+            // Combine the start and end sections
+            let mut start_section_vec = start_section.to_vec();
+
+            start_section_vec.extend_from_slice(&end_section);
+
+            bytecode = Bytes::from(start_section_vec);
```

Comment:
we should account for bytecode being shifted on every iteration, thus requiring next offsets to be shifted too